pkg/ingest/setup_test.go

```diff
@@ -3,11 +3,94 @@ package ingest
+		TaggedVersion: nonReleaseTagValue,
+	}
+	var sampleConfigIngested = v1.BuildRun{
+		Spec: &v1.BuildSpec{
+			Config: &v1.BuildConfig{
+				ReleaseContainerRegistry: &v1.ContainerRegistry{
+					Registry: &v1.ContainerRegistry_Quay{
+						Quay: &v1.QuayRegistry{
+							BaseUrl:      "quay.io",
+							Organization: "solo-io",
+						},
+					},
+				},
+				TestContainerRegistry: &v1.ContainerRegistry{
+					Registry: &v1.ContainerRegistry_Gcr{
+						Gcr: &v1.GoogleContainerRegistry{
+							BaseUrl:   "gcr.io",
+							ProjectId: "solo-public-1010",
+						},
+					},
+				},
+			},
+		},
+		Config: &v1.BuildRunConfig{
+			BuildEnvVars: &v1.BuildEnvVars{
+				BuildId:       buildId,
+				TaggedVersion: nonReleaseTagValue,
+			},
+			ComputedBuildVars: &v1.ComputedBuildVars{
+				Release:         false,
+				ImageTag:        buildId,
+				ContainerPrefix: "gcr.io/solo-public-1010",
+				Version:         buildId,
+			},
+		},
+	}
+
+	Context("InitializeBuildRun", func() {
+		It("should fallback to default filename when no file specified and env var not set", func() {
+			br, err := InitializeBuildRun("", &v1.BuildEnvVars{})
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("could not parse build spec: open solo-project.yaml: no such file or directory"))
+			Expect(br).To(Equal(v1.BuildRun{}))
+		})
+		It("should read from file when provided", func() {
+			br, err := InitializeBuildRun(relativePathToSampleConfig, explicitBuildEnvVars)
+			Expect(err).ToNot(HaveOccurred())
+
+			// aside on matching large structs
+			// Gomega has MatchAllFields, but it requires you to fill in everything, including the XXXabcd fields made by protoc
+			// An alternative is to pass IgnoreExtras with MatchFields
+			// Either way, you have to write out a lot of stuff by hand
+			// This is worthwhile in some cases because you can be explicit about how fields should match and you get
+			// great, scoped feedback on errors
+			Expect(br).To(MatchFields(IgnoreExtras, Fields{
+				"Spec": PointTo(MatchFields(IgnoreExtras, Fields{
+					"Config": PointTo(MatchFields(IgnoreExtras, Fields{
+						"ReleaseContainerRegistry": PointTo(MatchFields(IgnoreExtras, Fields{
+							"Registry": PointTo(MatchFields(IgnoreExtras, Fields{
+								"Quay": PointTo(MatchFields(IgnoreExtras, Fields{
+									"BaseUrl":      Equal("quay.io"),
+									"Organization": Equal("solo-io"),
+								})),
+							})),
+						})),
+					})),
+				})),
+			}))
+			// It is much easier to do this - stringify the struct
+			// One shortcoming is that you only get +/- 5 chars of context
+			// per: https://github.com/onsi/gomega/blob/master/format/format.go#L146
+			// TODO(mitchdraft) gomega pr to modify charactersAroundMismatchToInclude (if not merged will make a util)
+			Expect(br.String()).To(Equal(sampleConfigIngested.String()))
+			// This was my first thought, but the error message is not human friendly, you have to compare large blocks of text
```

Comment:
Is there a utility that could be written here for better diffing in the event of a failed expect (similar to your string context case)? This is the optimal test for readability

Reply:
let me know what you think of ExpectEqualProtoMessages, I can move it to go-utils if it looks good